pkg/sdk/testint/managed_accounts_gen_integration_test.go

```diff
@@ -0,0 +1,136 @@
+package testint
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_ManagedAccounts(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	assertManagedAccount := func(t *testing.T, managedAccount *sdk.ManagedAccount, id sdk.AccountObjectIdentifier, comment string) {
+		t.Helper()
+		assert.Equal(t, id.Name(), managedAccount.Name)
+		assert.Equal(t, "aws", managedAccount.Cloud)
+		assert.NotEmpty(t, managedAccount.Region)
+		assert.NotEmpty(t, managedAccount.Locator)
+		assert.NotEmpty(t, managedAccount.CreatedOn)
+		assert.NotEmpty(t, managedAccount.URL)
+		assert.NotEmpty(t, managedAccount.AccountLocatorURL)
+		assert.Equal(t, true, managedAccount.IsReader)
+		assert.Equal(t, comment, managedAccount.Comment)
+	}
+
+	cleanupMangedAccountProvider := func(id sdk.AccountObjectIdentifier) func() {
+		return func() {
+			err := client.ManagedAccounts.Drop(ctx, sdk.NewDropManagedAccountRequest(id))
+			require.NoError(t, err)
+		}
+	}
+
+	createManagedAccountBasicRequest := func(t *testing.T) *sdk.CreateManagedAccountRequest {
```

Comment:
we should have a discussion on what the structure for these kinds of test request helpers should be. i think there are a few different patterns right now. also there's the helpers_test.go file in the integration test folder, I think we should refactor that as well.

Reply:
We have already an issue for the helper test (SNOW-936093). We will have the discussion there. You are right we should unify the current approaches.